src/rarity/rarity.meta.ts

```diff
@@ -43,40 +44,52 @@ function hexToDec(s: string) {
         }
     }
     return digits.reverse().join('');
-}
+};
 
 export const stringToKeccak256DecimalId = (s: string, digits: number) =>
     hexToDec(keccak256(`0000${s}`.slice(-digits)).toString('hex'));
 
-const ensMetaFunc = (nftTraits: TraitBase[], collection: EnsCollectionSlug) => {
-    const max = ensCollectionSizes[collection as EnsCollectionSlug];
-    const digits = max.toString().length - 1;
-    const outTraits: TraitBase[] = [];
+const isEven = (x: number) => x % 2 === 0;
 
+const ensMetaFunc = (nftTraits: TraitBase[], collection: EnsCollectionSlug) => {
+    // Early exit
     const trait = nftTraits.find(t => t.displayType === 'number' && t.typeValue === ID_TRAIT_TYPE);
     if (!trait)
         throw new Error(`This trait needs a trait with a displayType of 'number' and a typeValue of ${ID_TRAIT_TYPE}`);
 
-    const i = +trait.value;
-    const stringified = `000${i}`.slice(-digits);
+    // Needed vars
+    const max = ensCollectionSizes[collection as EnsCollectionSlug];
+    const digits = max.toString().length - 1;
+    const id = +trait.value;
+    const stringifiedId = `000${id}`.slice(-digits);
+    const isEvenDigits = isEven(digits);
+    const outTraits: TraitBase[] = [];
 
     // Palindrome trait
     let isPalindrome = true;
-    if (digits === 4) {
+    if (isEvenDigits) {
+        // Even - Check for AABB
         let a = 0;
-        let b = stringified.length - 1;
+        let b = stringifiedId.length - 1;
         while (a < b) {
-            if (stringified[a] !== stringified[b]) {
+            if (stringifiedId[a] !== stringifiedId[b]) {
                 isPalindrome = false;
                 break;
             }
             a++;
             b--;
         }
-    } else if (digits === 3) {
-        isPalindrome = stringified[0] === stringified[2];
+    } else {
+        // Odd - Check for ABA or ABABA
+        const evenIndexValue = stringifiedId[0];
+        const oddIndexValue = stringifiedId[1];
+        for (let i = 2; i < digits; i++) {
+            isPalindrome = isEven(i) ? evenIndexValue === stringifiedId[i] : oddIndexValue === stringifiedId[i];
```

Comment:
would this catch "AABAA"? I think the first check, when i=2, would fail for AABAA